### PR TITLE
[17.0][MIG] sale_order_invoicing_finished_task: Migration to version 17.0

### PR DESCRIPTION
**Pull request by juanjosesegui-tecnativa**
@Tecnativa TT50120

### CODE REVIEW

**Comment by juanjosesegui-tecnativa** [comment]
ping @pedrobaeza

File: sale_order_invoicing_finished_task/models/project.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                self._check_sale_line_state(vals["sale_line_id"])
+        return super().create(vals_list)
+
+    def _check_sale_line_state(self, sale_line_id=False):
+        sale_lines = self.mapped("sale_line_id")
+        if sale_line_id:
+            sale_lines |= self.env["sale.order.line"].browse(sale_line_id)
+        for sale_line in sale_lines:
+            if (
+                sale_line.state in ("done", "cancel")
```
Comment (carlos-lopez-tecnativa) [inline sale_order_invoicing_finished_task/models/project.py]: The option `done` no longer exists in the `sale.order` field `state`(which is related to `sale.order`) , please use the `sale` option or the `locked` field according to your business needs.
https://github.com/odoo/odoo/blob/279f0566640341dbb8305a8e4e15784819586c5c/addons/sale/models/sale_order.py#L24

**Review by carlos-lopez-tecnativa — CHANGES_REQUESTED** [review]

File: sale_order_invoicing_finished_task/views/product_view.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            name="inherit_id"
+            ref="sale_project.product_template_form_view_invoice_policy_inherit_sale_project"
+        />
+        <field name="arch" type="xml">
+            <field name="project_id" position="before">
+                <field
+                    name="invoicing_finished_task"
+                    invisible="type != 'service' or service_tracking not in ['task_global_project', 'task_in_project']"
+                />
+            </field>
```
Comment (carlos-lopez-tecnativa) [inline sale_order_invoicing_finished_task/views/product_view.xml]: Improve UI
image
With option `Create Order = Task` selected
image

**Review by carlos-lopez-tecnativa — CHANGES_REQUESTED** [review]
A tiny visual change.

File: sale_order_invoicing_finished_task/views/product_view.xml (hunk 2 above)
Comment (juanjosesegui-tecnativa) [reply sale_order_invoicing_finished_task/views/product_view.xml]: I have made the changes, and it is now displaying correctly.

Comment (carlos-lopez-tecnativa) [reply sale_order_invoicing_finished_task/views/product_view.xml]: Hi, sorry but i not see the field, can you please check it
image

File: sale_order_invoicing_finished_task/views/product_view.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        <field name="model">product.template</field>
+        <field
+            name="inherit_id"
+            ref="sale_project.product_template_form_view_invoice_policy_inherit_sale_project"
+        />
+        <field name="arch" type="xml">
+            <field name="project_id" position="before">
+                <field
+                    name="invoicing_finished_task"
+                    invisible="(type, '!=', 'service') or (service_tracking not in ['task_global_project', 'task_in_project'])"
```
Comment (carlos-lopez-tecnativa) [inline sale_order_invoicing_finished_task/views/product_view.xml]: This expression is wrong, but I think it does not address my comment. It is more related to where the field is added to the view, not the invisible attribute.

```suggestion
                    invisible="(type != 'service') or (service_tracking not in ['task_global_project', 'task_in_project'])"
```

Comment (juanjosesegui-tecnativa) [reply sale_order_invoicing_finished_task/views/product_view.xml]: Changes are done, take a look when you can.

**Comment by juanjosesegui-tecnativa** [comment]
Changes are done, take a look when you can.

@carlos-lopez-tecnativa

**Review by carlos-lopez-tecnativa — APPROVED** [review]
LGTM, thanks

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**